.rustfmt.toml

```diff
@@ -0,0 +1,16 @@
+ignore = [
+  "quickwit-proto/src/cluster.rs",
+  "quickwit-proto/src/quickwit.rs",
+  "quickwit-swim",
+]
+
+comment_width = 120
+format_strings = true
+group_imports = "StdExternalCrate"
+imports_granularity = "Module"
```

Comment:
I think there is a real downside on increased merge conflicts and would not automatically group imports for that reason

Reply:
Those conflicts are super easy to resolve though: 
1. remove the git markers;
2. run `fmt` again to get rid of duplicates;
3. in some cases, remove the unused imports.

I personally find that this overhead is preferable over hard-to-read nested imports and contributors arguing over how imports should be organized:
```rust
use foo::{self, bar::{self, qux::{self, ...}}
};
```
In my opinion, the more code formatting can be automated, the better.

Reply:
It's still an extra step where you need to check if it's a semantic merge conflict or only imports.

I agree on more formatting is better, since it usually increases readability and reduces merge conflicts.
But in my experience this is not true for imports since readability on imports is not really important and merge conflicts can happen very frequent with that setting.

In practice I think our imports are just a flat list and not such a nested structure, since they are hard to type and maintain.

Reply:
Indeed, merge conflicts coming from formatting are annoying. Currently, I did not really have to handle formatting merge conflicts or it was really minor. I don't think formatting the imports will change much so I think we can test it and rethink about it as the project grows.